pymc3/gp/gp.py

```diff
@@ -106,18 +108,18 @@ def sample_gp(trace, gp, X_values, samples=None, obs_noise=True, model=None, ran
     -------
     Array of samples from posterior GP evaluated at Z.
     """
-    model = modelcontext(model)
-
     if samples is None:
         samples = len(trace)
 
+    if model is None:
```

Comment:
I think there is no need for None check, `modelcontext` [does this already](https://github.com/pymc-devs/pymc3/blob/master/pymc3/model.py#L144)